autoendpoint/src/server/routers/webpush.rs

```diff
@@ -0,0 +1,217 @@
+        trace!("Notifying node to check for messages");
+        match self.trigger_notification_check(&user.uaid, &node_id).await {
+            Ok(response) => {
+                trace!("Response = {:?}", response);
+                if response.status() == 200 {
+                    trace!("Node has delivered the message");
+                    Ok(self.make_delivered_response(notification))
+                } else {
+                    trace!("Node has not delivered the message, returning stored response");
+                    Ok(self.make_stored_response(notification))
+                }
+            }
+            Err(error) => {
+                // Can't communicate with the node, so we should stop using it
+                debug!("Error while triggering notification check: {}", error);
+                self.remove_node_id(&user, node_id.clone()).await?;
+                Ok(self.make_stored_response(notification))
+            }
+        }
+    }
+}
+
+impl WebPushRouter {
+    /// Send the notification to the node
+    async fn send_notification(
+        &self,
+        notification: &Notification,
+        node_id: &str,
+    ) -> Result<Response, reqwest::Error> {
+        let url = format!("{}/push/{}", node_id, notification.subscription.user.uaid);
+        let notification = notification.serialize_for_delivery();
+
+        self.http.put(&url).json(&notification).send().await
+    }
+
+    /// Notify the node to check for notifications for the user
+    async fn trigger_notification_check(
+        &self,
+        uaid: &Uuid,
+        node_id: &str,
+    ) -> Result<Response, reqwest::Error> {
+        let url = format!("{}/notif/{}", node_id, uaid);
+
+        self.http.put(&url).send().await
+    }
+
+    /// Store a notification in the database
+    async fn store_notification(&self, notification: &Notification) -> ApiResult<()> {
+        self.ddb
+            .store_message(
+                &notification.subscription.user.uaid,
+                notification
+                    .subscription
+                    .user
+                    .current_month
+                    .clone()
+                    .unwrap_or_else(|| self.ddb.current_message_month.clone()),
+                notification.clone().into(),
+            )
+            .compat()
+            .await
+            .map_err(|e| ApiErrorKind::Router(RouterError::SaveDb(e)).into())
+    }
+
+    /// Remove the node ID from a user. This is done if the user is no longer
+    /// connected to the node.
+    async fn remove_node_id(&self, user: &DynamoDbUser, node_id: String) -> ApiResult<()> {
+        self.metrics.incr("updates.client.host_gone").ok();
+
+        self.ddb
+            .remove_node_id(&user.uaid, node_id, user.connected_at)
+            .compat()
+            .await
+            .map_err(|e| ApiErrorKind::Database(e).into())
+    }
+
+    /// Update metrics and create a response for when a notification has been directly forwarded to
```

Comment:
Not like it's a huge deal, but why the intermediary function? It's adding a few static args. 

"Aesthetics" is a reasonable answer.

Reply:
1. The function is used multiple times and this avoids duplicating those static args. Now we only need to change the args in one place.
2. The Python code had this pattern, and it seemed reasonable enough to reuse. One difference is that the Python code duplicated the metrics and response object builder code, which I've moved into `make_response`.